instagram-app/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import Fuse from "fuse.js";
+import data from "../../dummy-data";
+import SeachBar from "../SearchBar/SearchBar";
+import PostContainer from "./PostContainer";
+
+export default class PostsPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      posts: [],
+      searchInput: ""
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("beforeunload", () => {
+      this.updateLocalStorage();
+    });
+
+    let initialPosts;
+
+    try {
+      initialPosts = JSON.parse(localStorage.instagramClone);
+    } catch (e) {
+      initialPosts = data;
+    }
+
+    this.setState({
+      posts: initialPosts
+    });
+  }
+
+  likeComment = id => {
+    const newPosts = [...this.state.posts];
+
+    newPosts.forEach(post => {
+      if (post.id === id) {
+        if (!post.liked) {
+          post.likes += 1;
+          post.liked = !post.liked;
+        } else {
+          post.likes -= 1;
+          post.liked = !post.liked;
+        }
+      }
+    });
+
+    this.setState({
+      posts: newPosts
+    });
+  };
+
+  searchInputChange = e => {
+    this.setState({
+      searchInput: e.target.value
+    });
+  };
+
+  updateLocalStorage = () => {
+    localStorage.instagramClone = JSON.stringify(this.state.posts);
+  };
+
+  render() {
+    const { posts, searchInput } = this.state;
+    const options = {
+      keys: ["username"]
+    };
+
+    // eslint-disable-next-line
+    const postContainers = posts.map(post => {
```

Comment:
I'm not a fan of this component declared here inside your render function. It's probably better to have it in his separate module, or at least as a method of the class.